Azure AD PowerShell to Microsoft Graph PowerShell migration FAQ (#768)
Azure/azure-docs-powershell-azuread: docs-conceptual/azureadps-2.0/migration-faq.md

---
title: 'Azure AD PowerShell to Microsoft Graph PowerShell Migration FAQ'
description: Answers to frequently asked questions about the migration from Azure AD PowerShell to Microsoft Graph PowerShell.
ms.service: active-directory
ms.workload: identity
ms.topic: article
ms.date: 01/27/2022
ms.author: eunicewaweru
---

# Azure AD PowerShell to Microsoft Graph PowerShell migration FAQ

This article provides answers to frequently asked questions about migrating from Azure AD PowerShell to Microsoft Graph PowerShell.

## What happens to Azure AD PowerShell come June 2022?

> In June 2020, we announced the retirement of Azure AD Graph API in favor of the Microsoft Graph API. As a result, all applications calling this API would receive an error message from the endpoint come this date. Azure AD PowerShell is one of the applications that would be affected by this retirement. After analysis and talking to our customers, we have realized that reliance on this endpoint is still high. After much deliberation, we've opted not to retire the endpoint come June 2022 to allow users more time to migrate to [Microsoft Graph PowerShell](/powershell/microsoftgraph/overview). We will take a phased shutdown approach guided by customer data on the rate of migration to Microsoft Graph PowerShell. This data will guide the process of retiring key functionalities such as the ability to create new applications on the portal. Come June 2022, we will stop creation of Azure AD Graph based apps. The aim is to allow a smooth customer migration onto Microsoft Graph PowerShell with a graceful shutdown of the Azure AD Graph endpoint. Users should continue to use Azure AD PowerShell for the next couple of years but we encourage you to continue migrating to Microsoft Graph PowerShell SDK.  

Reviewer: ```suggestion
In June 2020, we announced the retirement of Azure AD Graph API in favor of the Microsoft Graph API. As a result, all applications calling this API would receive an error message from the endpoint come this date. Azure AD PowerShell is one of the applications that would be affected by this retirement. After analysis and talking to our customers, we realized that reliance on this endpoint is still high. After much deliberation, we've opted not to retire the endpoint come June 2022 to allow users more time to migrate to [Microsoft Graph PowerShell](/powershell/microsoftgraph/overview). We will take a phased shutdown approach guided by customer data on the rate of migration to Microsoft Graph PowerShell. This data will guide the process of retiring key functionalities including the ability to create new applications on the Azure portal. Come June 2022, we will stop creation of Azure AD Graph-based apps. The aim is to allow a smooth customer migration to Microsoft Graph PowerShell with a graceful shutdown of the Azure AD Graph endpoint. Users can continue to use Azure AD PowerShell for the next two years, but we encourage you to migrate to Microsoft Graph PowerShell SDK.  
```

## What are we doing about the gaps in functionality between Azure AD and Microsoft Graph PowerShell modules?

> As part of the parity work, we are looking at MSOL (MSOnline) and Azure AD PowerShell  glossary of cmdlets and mapping the functionality onto Microsoft Graph PowerShell. This is to make sure that all old scenarios are available in Microsoft Graph PowerShell. We have started working on [a cmdlet map](/powershell/microsoftgraph/azuread-msoline-cmdlet-map) for both MSonline and Azure AD PowerShell cmdlets to Microsoft Graph PowerShell, as part of the migration documentation. Users have also reached out with specific scenarios and cmdlets that we will add to Microsoft Graph PowerShell. We are also working on tools such as the [Find-Mg command](/powershell/microsoftgraph/find-mg-graph-command) that will assist users in discovering the appropriate Microsoft Graph PowerShell cmdlets from a command name or a Microsoft Graph URI.  

Reviewer: ```suggestion
As part of the parity work, we're looking at an MSOnline (MSOL) and Azure AD PowerShell glossary of cmdlets and mapping the functionality to Microsoft Graph PowerShell. This is to make sure that all existing scenarios are available in Microsoft Graph PowerShell. We started working on [a cmdlet map](/powershell/microsoftgraph/azuread-msoline-cmdlet-map) for both MSOnline and Azure AD PowerShell cmdlets to Microsoft Graph PowerShell as part of the migration documentation. Users have also reached out with specific scenarios and cmdlets that we will add to Microsoft Graph PowerShell. We're also working on tools such as the [Find-Mg command](/powershell/microsoftgraph/find-mg-graph-command) that will assist users in discovering the appropriate Microsoft Graph PowerShell cmdlets from a command name or a Microsoft Graph URI.  
```

## What are we doing to improve the Microsoft Graph PowerShell user experience?

> Microsoft Graph PowerShell cmdlets are autogenerated from the Microsoft Graph API. The process has its advantages around having a rich set of cmdlets that cover the whole API landscape. The disadvantage is that cmdlets are not user-friendly compared to human authored cmdlets. We have embarked on improving these user experience (UX) gaps by building other friendlier cmdlets that borrow a lot from Azure AD Graph PowerShell. The objective is to wrap the autogenerated cmdlets with custom usecase-driven cmdlets that make it easy for the users to access Azure AD services.  

Reviewer: ```suggestion
Microsoft Graph PowerShell cmdlets are autogenerated from the Microsoft Graph API. The process has its advantages around having a rich set of cmdlets that cover the whole API landscape. The disadvantage is that cmdlets are not user-friendly compared to human authored cmdlets. We are improving these user experience (UX) gaps by building other friendlier cmdlets that borrow a lot from Azure AD Graph PowerShell. The objective is to wrap the autogenerated cmdlets with custom use case-driven cmdlets that make it easy for users to access Azure AD services.  
```

The Microsoft Graph PowerShell SDK is also open source and users can contribute to the UX work or add new scenarios to manage Azure AD.

> ## How are we going to support our users on migrating to Microsoft Graph PowerShell?

Reviewer: ```suggestion
## How are we going to support users who are migrating to Microsoft Graph PowerShell?
```

> We are working on a migration tool that will incorporate all existing Azure AD Graph cmdlets in a module powered by the Microsoft Graph API. The module borrows the cmdlet structure and return type from the Azure AD Graph PowerShell. Users will download this tool as a separate module solely for migration.  

Reviewer: ```suggestion
We're working on a migration tool that will incorporate all existing Azure AD Graph cmdlets in a module powered by the Microsoft Graph API. The module borrows the cmdlet structure and return type from the Azure AD Graph PowerShell. Users will be able to download this tool as a separate module they can use solely for migration.  
```

> The module can be used in two main ways: search and replace the Azure AD cmdlet prefix with the new one in scripts or optionally enable aliases of original commands to run scripts unmodified. More information on this module will be communicated soon.

Reviewer: ```suggestion
The module can be used in two main ways:
- Search and replace the Azure AD cmdlet prefix with the new one in scripts, or 
- Optionally enable aliases of original commands to run scripts unmodified. 

We will provide more info about this module soon.
```
